content/en/docs/architecture/ci-operator-internals/artifacts.md

```diff
@@ -0,0 +1,163 @@
+    ],
+    "bucket": "origin-ci-test",
+    "path_strategy": "single",
+    "default_org": "openshift",
+    "default_repo": "origin",
+    "mediaTypes": {
+      "log": "text/plain"
+    },
+    "gcs_credentials_file": "/secrets/gcs/service-account.json",
+    "dry_run": false
+  },
+  "entries": [
+    {
+      "args": [
+        "ci-operator",
+        "--gcs-upload-secret=/secrets/gcs/service-account.json",
+        "--image-import-pull-secret=/etc/pull-secret/.dockerconfigjson",
+        "--report-credentials-file=/etc/report/credentials",
+        "--target=gofmt"
+      ],
+      "container_name": "test",
+      "process_log": "/logs/process-log.txt",
+      "marker_file": "/logs/marker-file.txt",
+      "metadata_file": "/logs/artifacts/metadata.json"
+    }
+  ],
+  "censoring_options": {
+    "secret_directories": [
+      "/etc/pull-secret",
+      "/etc/report"
+    ]
+  }
+}
+{{< / highlight >}}
+
+- `gcs_options` control how and where in GCS files are placed.  The final paths
+  are determined by these options and the Prow job's `JOB_SPEC`.
+
+  - Files from a periodic or post-submit job called `$name` with build ID
+    `$build_id` can be listed with:
+
+    {{< highlight console >}}
+$ gsutil ls gs://$bucket/logs/$name/$build_id/artifacts/
+    {{< / highlight >}}
+  - Files from a pre-submit job for pull request `$pull` in repository
+    `$org/$repo` can be listed with:
+
+    {{< highlight console >}}
+$ gsutil ls gs://$bucket/pr-logs/pull/$org_$repo/$pull/$name/$build_id/artifacts/
+    {{< / highlight >}}
+
+- `entries` contains information about test processes to wait for and collect
+  data from.  These correspond to the `entrypoint` options.
+
+- `censoring_options` controls the censoring of artifacts prior to the upload
+  based on the content of secrets mounted into the `Pod`.
+
+## Testing
+
+When executing `ci-operator` locally, most of the time artifacts are not
+necessary.  If that is the case, simply not including a `gcs_credentials_secret`
+(usually set to `gce-sa-credentials-gcs-publisher`) in the `JOB_SPEC` will allow
+tests to be executed.  The `sidecar` container will fail to upload artifacts,
+but that is not considered an error by `ci-operator`.  Alternatively, creating
+an empty `Secret` object in the test namespace (either directly or via
+`ci-operator`'s `--secret-dir` argument) will cause `sidecar` to fail
+immediately and not even attempt to gather or upload artifacts.
+
+For some test setups, artifact gathering is required, and `ci-operator` does not
+currently provide a way to disable it.  In order to executed tests locally, the
+test namespace needs to contain valid GCS credentials.  In this case, or if
+artifacts actually need to be inspected, the `Secret` can be copied from one of
+the build clusters:
+
+{{< highlight console >}}
+$ oc create secret generic gce-sa-credentials-gcs-publisher --from-literal "service-account.json=$(oc -n ci extract --to - secret/gce-sa-credentials-gcs-publisher)"
```

Comment:
Please split this line in (at least) two parts. The page format results to be broken otherwise.